### ISSUE TITLE
Installation issue: postgresql and multiple build systems

### ISSUE THREAD

**Issue by glennpj** [LLNL/spack#33808]
### Steps to reproduce the issue

```console
$ spack install postgresql
...
```

### Error message

<details><summary>Error message</summary><pre>
==> Error: AttributeError: type object 'AutotoolsPackage' has no attribute 'install'

The 'postgresql' package cannot find an attribute while trying to build from sources. This might be due to a change in Spack's package format to support multiple build-systems for a single package. You can fix this by updating the build recipe, and you can also report the issue as a bug. More information at https://spack.readthedocs.io/en/latest/packaging_guide.html#installation-procedure

/opt/packages/gpjohnsn/spack/var/spack/repos/builtin/packages/postgresql/package.py:103, in install:
        100                with working_dir(os.path.join("src", subdir)):
        101                    make("install")
        102        else:
  >>    103            AutotoolsPackage.install(self, spec, prefix)

</pre></details>

### Information on your system

* **Spack:** 0.19.0.dev0
* **Python:** 3.6.8
* **Platform:** linux-centos7-cascadelake
* **Concretizer:** clingo

### Additional information

@alalazo Here is another one that fails after the multi build system PR.

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

**Comment by alalazo** [comment]
Fixed by #33770

**Comment by glennpj** [comment]
Oh, sorry; I somehow missed that.

**Comment by alalazo** [comment]
No worries. As an explanation for the cryptic error, in #30738 we moved methods from one class hierarchy to another. 

`AutotoolsPackage` doesn't have anymore an `install` method, so if people call the method on a specific class, like `AutotoolsPackage.install`, they won't find it. To maintain an almost complete backward compatibility with the recipes we've got in the `builtin` repository, I coded a few adapters that are auto-generated by Spack. Bottom line is that those adapters can route a call to `super(X, self)` to find methods that where previously on X and have now been moved to some other class.
